gdraw/gfilechooser.c

```diff
@@ -460,6 +486,26 @@ static void GFileChooserScanDir(GFileChooser *gfc,unichar_t *dir) {
 	ti = malloc((cnt+1)*sizeof(GTextInfo *));
 	tot = cnt-1;
     }
+
+#ifdef _WIN32
+    ti = realloc(ti, (dcnt+cnt+1)*sizeof(GTextInfo *));
+    for (char* c = drives; *c != '\0'; c++) {
+        // Don't put the root twice if we're already on it
+        unichar_t buffer[1025];
+        if (*c == toupper(u_GFileGetAbsoluteName(dir,buffer,sizeof(buffer)/sizeof(unichar_t))[0])) continue;
+
+        ti[cnt] = calloc(1,sizeof(GTextInfo));
+        unichar_t* utemp = calloc(3,sizeof(unichar_t));
+        utemp[0] = *c; utemp[1] = ':'; //utemp[2] = '\\'; // utemp[3] = 0;
+        ti[cnt]->text = utemp;
+        ti[cnt]->fg = ti[cnt]->bg = COLOR_DEFAULT;
+        // We don't want to add a new field to GTextInfo as it's used in a lot of places.
+        // This is the next best thing AFAICT.
+        ti[cnt]->userdata = (void*)1;
```

Comment:
I'm not asking for a full review @skef because I know you don't use Windows. I just wanted to get your opinion on these lines if you have a moment. Is this `(void*)1` acceptable in your opinion? `GTextInfo` literals are all over the codebase, so to add a field would be, I think, (without checking!) around a +150,-150 or more change.